internal/testutil/mig/provider.go

```diff
@@ -0,0 +1,61 @@
+package mig
+
+import (
+	"os"
+	"testing"
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+)
+
+func ExternalProviders(tb testing.TB) map[string]resource.ExternalProvider {
+	checkLastVersion(tb)
+	return map[string]resource.ExternalProvider{
+		"mongodbatlas": {
+			VersionConstraint: versionConstraint(),
+			Source:            "mongodb/mongodbatlas",
+		},
+	}
+}
+
+func ExternalProvidersWithAWS(tb testing.TB, awsVersion string) map[string]resource.ExternalProvider {
+	checkLastVersion(tb)
+	return map[string]resource.ExternalProvider{
+		"mongodbatlas": {
+			VersionConstraint: versionConstraint(),
+			Source:            "mongodb/mongodbatlas",
+		},
+		"aws": {
+			VersionConstraint: awsVersion,
+			Source:            "hashicorp/aws",
+		},
+	}
+}
+
+func ValueIfVersionAtLeast[T any](tb testing.TB, val T, minVersion string) T {
```

Comment:
```suggestion
func ConfiguredProviderVersionAtLeast(tb testing.TB,  minVersion string) bool
```
nit: might find it easier to follow if this function returns a boolean, and then each function that generates a configuration uses that boolean to their preference (passing default value, or using the boolean itself to define a set of attributes in the config).

Reply:
makes sense, changed